term-utils/agetty.c

```diff
@@ -1968,22 +1977,23 @@ static void eval_issue_file(struct issue *ie,
 		goto done;
 	}
 
-	/* The default /etc/issue and optional /etc/issue.d directory as
-	 * extension to the file. The /etc/issue.d directory is ignored if
-	 * there is no /etc/issue file. The file may be empty or symlink.
+        /* Reading all issue files and concatinating all contents to one content.
+         * The ordering rules are defineded in:
+         * https://github.com/uapi-group/specifications/blob/main/specs/configuration_files_specification.md
 	 */
-	if (access(_PATH_ISSUE, F_OK|R_OK) == 0) {
-		issuefile_read(ie, _PATH_ISSUE, op, tp);
-		issuedir_read(ie, _PATH_ISSUEDIR, op, tp);
-	}
-
-	/* Fallback @runstatedir (usually /run) */
-	issuefile_read(ie, _PATH_RUNSTATEDIR "/" _PATH_ISSUE_FILENAME, op, tp);
-	issuedir_read(ie, _PATH_RUNSTATEDIR "/" _PATH_ISSUE_DIRNAME, op, tp);
-
-	/* Fallback @sysconfstaticdir (usually /usr/lib)*/
-	issuefile_read(ie, _PATH_SYSCONFSTATICDIR "/" _PATH_ISSUE_FILENAME, op, tp);
-	issuedir_read(ie, _PATH_SYSCONFSTATICDIR "/" _PATH_ISSUE_DIRNAME, op, tp);
+	config_file_list (&file_list,
+			  NULL,
+			  _PATH_ETC_ISSUEDIR,
+			  _PATH_USR_ISSUEDIR,
+			  _PATH_ISSUE_FILENAME,
+			  ISSUEDIR_EXT);
+
+	list_for_each(file_entry, &file_list) {
+		struct file_element *element;
+	        element = list_entry(file_entry, struct file_element, file_list);
+		issuefile_read(ie, element->filename, op, tp);
+	}
+	list_free(&file_list, struct file_element,  file_list, free_element);
 
```

Comment:
It would be better to introduce `free_config_file_list()` rather than assuming that we will define `static free_element()` in all utils and call it via list_free().